modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java

```diff
@@ -109,16 +110,10 @@ private void openStreamWithRetry() throws IOException {
                     }
                     if (amazonS3Exception.getStatusCode() == RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()) {
                         throw addSuppressedExceptions(
-                            new IOException(
-                                "Requested range [start="
-                                    + start
-                                    + ", end="
-                                    + end
-                                    + ", currentOffset="
-                                    + currentOffset
-                                    + "] cannot be satisfied for blob object ["
-                                    + blobKey
-                                    + ']',
+                            new RequestedRangeNotSatisfiedException(
```

Comment:
It is unchanged with this PR, but I wonder if we could see a failure on getting the EOF marker and whether that could lead to a read beyond the blob length? Nevertheless, this will still work for us I think.

Reply:
I initially thought there could be subtleties here (like the retrying stream making some progress and then failing when reading the very last byte of the blob) but I think it will work: all callers of that method should request ranges that are valid and exist, except for the case that motivated the specialized exception and for which we should handle the exception appropriately.